format/format.go

```diff
@@ -317,22 +317,33 @@ func (f *fumpter) applyPre(c *astutil.Cursor) {
 	case *ast.File:
 		// Join contiguous lone var/const/import lines.
 		// Abort if there are empty lines or comments in between,
-		// including a leading comment, which could be a directive.
+		// including a leading comment if it's a directive.
 		newDecls := make([]ast.Decl, 0, len(node.Decls))
 		for i := 0; i < len(node.Decls); {
 			newDecls = append(newDecls, node.Decls[i])
 			start, ok := node.Decls[i].(*ast.GenDecl)
-			if !ok || isCgoImport(start) || start.Doc != nil {
+			if !ok || isCgoImport(start) || isCommentGrpDirective(start.Doc) {
 				i++
 				continue
 			}
 			lastPos := start.Pos()
+		contLoop:
 			for i++; i < len(node.Decls); {
 				cont, ok := node.Decls[i].(*ast.GenDecl)
-				if !ok || cont.Tok != start.Tok || cont.Lparen != token.NoPos ||
-					f.Line(lastPos) < f.Line(cont.Pos())-1 || isCgoImport(cont) {
+				if !ok || cont.Tok != start.Tok || cont.Lparen != token.NoPos || isCgoImport(cont) {
 					break
 				}
+				if f.Line(lastPos) < f.Line(cont.Pos())-1 {
```

Comment:
I think `f.Line(lastPos) < f.Line(cont.Pos())-1` can be translated to:
are there things between these two declarations? e.g. empty lines, comments, directives

and `containsAnyDirective` is self explanatory

so `break` on `f.Line(lastPos) < f.Line(cont.Pos())-1 && containsAnyDirective(cont.Doc)`  would break the chain only on directives but would leave the chain going on empty lines

Reply:
I see, good point. perhaps add a bit of docs as at least I misunderstood what we meant the code to do.

Reply:
done, PTAL, thanks